### PR TITLE
Add text output (fixes #51)

### LINKED ISSUE #51: Add text output processing.
We've got a start on outputting JSON - we need to add something to output the rules in a nice way as just text with nice colours assuming boring is false.

### PR DESCRIPTION
**Pull request by muffinresearch**
(fixes #51)

Added the text ouput - whilst it's not crucial it'll be nice for development as this is easier to grok than looking at a pile of JSON.

1__bash

### CODE REVIEW

File: src/validator.js [hunk 1]
```diff
@@ -84,6 +100,59 @@ export default class Validator {
...
+    out.push(_('Validation Summary:'));
+    out.push('');
+    out.push(columnify(this.output.summary, {
+      showHeaders: false,
+      minWidth: 15,
+    }));
+    out.push('');
+
+    for (let type of constants.MESSAGE_TYPES) {
+      /*eslint-disable no-loop-func */
```
Comment (tofumatt) [inline src/validator.js]: We're using `let` so this shouldn't be an issue: http://eslint.org/docs/rules/no-loop-func.html#rule-details

Was this giving you an error? :frowning:

**Comment by tofumatt** [comment]
:+1: Very nice. Let me know about the eslint bit, but if it needs to be there that's fine. Maybe we even want to remove that rule (we can be smart about loops I think).

File: src/validator.js (hunk 1 above)
Comment (muffinresearch) [inline src/validator.js]: Yeah it's the callback funcs passed into then etc. I reckon you're right, we should probably just turn that rule off.

File: src/javascript.js [hunk 2]
```diff
@@ -33,7 +33,8 @@ export default class JavaScriptScanner {
           id: message.ruleId.toUpperCase(),
           message: messages[message.ruleId.toUpperCase()].message,
           description: messages[message.ruleId.toUpperCase()].description,
-          code: message.source,
+          code: message.ruleId.toUpperCase(),
```
Comment (tofumatt) [inline src/javascript.js]: We already use `id` for this, so should we ditch the `id` property? I'm fine with that.

File: src/javascript.js [hunk 3]
```diff
@@ -33,7 +33,8 @@ export default class JavaScriptScanner {
...
+          eslintCode: message.source,
```
Comment (tofumatt) [inline src/javascript.js]: This is the source code that caused the error, so a property name like `sourceCode` or `sourceText` would be better.

File: tests/test.messages.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import * as messages from 'messages';
+
+describe('Messages', function() {
+
+  it('should only have codes with a length < 25', () => {
```
Comment (tofumatt) [inline tests/test.messages.js]: That's a good test to have; prevents `E_TOO_LONG_ERROR_NAMES_THAT_ARE_A_BIT_OVERKILL` too. :+1:

**Comment by tofumatt** [comment]
This looks good; we should stop using `id` at all and switch over to `code` everywhere. I can do that if you like in a separate PR as housekeeping.

`eslintCode` should be something like `source`, `sourceCode`, or `sourceText`.

r+wc

File: src/javascript.js (hunk 2 above)
Comment (muffinresearch) [inline src/javascript.js]: Ok cool, I'll remove id.

File: src/javascript.js (hunk 3 above)
Comment (muffinresearch) [inline src/javascript.js]: Sounds good I'll use that.

**Pull request merged**